browser/playlist/playlist_page_handler.h

```diff
@@ -3,30 +3,39 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef BRAVE_BROWSER_UI_WEBUI_PLAYLIST_PAGE_HANDLER_H_
-#define BRAVE_BROWSER_UI_WEBUI_PLAYLIST_PAGE_HANDLER_H_
+#ifndef BRAVE_BROWSER_PLAYLIST_PLAYLIST_PAGE_HANDLER_H_
+#define BRAVE_BROWSER_PLAYLIST_PLAYLIST_PAGE_HANDLER_H_
 
 #include <string>
 
 #include "base/scoped_observation.h"
 #include "brave/components/playlist/mojom/playlist.mojom.h"
 #include "brave/components/playlist/playlist_service.h"
 #include "brave/components/playlist/playlist_service_observer.h"
+#include "components/keyed_service/core/keyed_service.h"
 #include "mojo/public/cpp/bindings/pending_receiver.h"
 #include "mojo/public/cpp/bindings/pending_remote.h"
 #include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/receiver_set.h"
 #include "mojo/public/cpp/bindings/remote.h"
 
 class Profile;
 
-class PlaylistPageHandler : public playlist::mojom::PageHandler,
+class PlaylistPageHandler : public KeyedService,
```

Comment:
Instead of making existing `PlaylistPageHandler` as a service,
how about creating new service and making that service own this handler?
this is shared with desktop but desktop doesn't need it as a searvice.